src/foraging_gui/Visualization.py

```diff
@@ -61,34 +73,43 @@ def _Update(self,GeneratedTrials=None,Channel=None):
 
         try:
             self._PlotBlockStructure()
+        except Exception as e:
+            logging.error(str(e))
+
+        try:
             self._PlotChoice()
+        except Exception as e:
+            logging.error(str(e))
+
+        try:
             self._PlotLicks()
         except Exception as e:
             logging.error(str(e))
+
         try:
             self._PlotMatching()
         except Exception as e:
             logging.error(str(e))
-        self.finish=1
```

Comment:
@XX-Yin what was the original purpose of setting a `finish` flag? Is it related to threading? Is it safe to remove?